crates/mongodb-agent-common/src/query/foreach.rs

```diff
@@ -56,7 +56,7 @@ pub fn pipeline_for_foreach(
     foreach: Vec<ForeachVariant>,
     query_request: &QueryRequest,
 ) -> Result<(Pipeline, ResponseShape), MongoAgentError> {
-    let pipelines_with_response_shapes: BTreeMap<String, (Pipeline, ResponseShape)> = foreach
+    let pipelines_with_response_shapes: Vec<(String, (Pipeline, ResponseShape))> = foreach
```

Comment:
This PR contains a lot of fixture and test changes. This line is the only change required for the actual fix.